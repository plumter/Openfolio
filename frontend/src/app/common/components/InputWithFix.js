

const InputWithFix = ({prefix, affix, inputRef, ...props}) => {

    return <div className={`relative my-auto`}>
                    {
                        prefix && 
                                    <div className="absolute inset-y-0 left-0 flex items-center px-2">
                                        <label 
                                            className=" rounded font-medium px-2 pt-2.5 text-sm dark:text-white cursor-pointer z-30" 
                                        >
                                            {prefix}
                                        </label>
                                    </div>
                    }
                <input 
                    className={`form-input ${prefix && "pl-14"} ${affix && "pr-14"}`}
                    type="text"
                    {...props}
                    ref={inputRef}
                />
                {
                    affix &&
                                <div className="absolute inset-y-0 right-0 flex items-center px-2 "> 
                                    <label 
                                        className=" font-medium rounded px-2 pt-2.5 text-sm text-gray-600 z-50" 
                                    >
                                        {affix}
                                    </label>
                                </div>
                }
            </div>
}

export default InputWithFix;