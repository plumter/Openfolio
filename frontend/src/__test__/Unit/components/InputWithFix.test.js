import InputWithFix from 'app/common/components/InputWithFix';
import { create } from 'react-test-renderer';



describe("InputWithFix Component Unit Testing", _ => {

    it('Test InputWithFix Cases',  () => {
      // 1st case
      const component = create(      
          <InputWithFix
              type='text' 
              placeholder="0.00" 
              pattern="[\d]{1,}"
              required 
              name="test1"
              key={1}
          />,
        );
      expect(component.toJSON()).toMatchSnapshot();

      // 2nd case
      component.update(      
          <InputWithFix
              type='amount' 
              placeholder="0.00" 
              pattern="[\d\.]{1,}"
              required 
              name="test2"
              prefix="$"
              key={2}
          />,
        );
      expect(component.toJSON()).toMatchSnapshot();

      // 3rd case
      component.update(      
          <InputWithFix
              type='text' 
              placeholder="0.00" 
              pattern="^(\d{0,2}(\.\d+)?|100(\.0+)?)$"
              required 
              name="percent"
              affix="%"
              key={3}
          />,
        );
      expect(component.toJSON()).toMatchSnapshot();

      // 4th case
      component.update(      
          <InputWithFix
              type='text' 
              placeholder="00" 
              pattern="[\d\.]{1,}"
              required 
              name="amount"
              prefix="$"
              affix=".00"
              key={4}
          />,
        );
      expect(component.toJSON()).toMatchSnapshot();
    });  
});