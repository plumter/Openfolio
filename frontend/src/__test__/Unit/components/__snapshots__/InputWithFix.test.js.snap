// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputWithFix Component Unit Testing Test InputWithFix Cases 1`] = `
<div
  className="relative my-auto"
>
  <input
    className="form-input undefined undefined"
    name="test1"
    pattern="[\\\\d]{1,}"
    placeholder="0.00"
    required={true}
    type="text"
  />
</div>
`;

exports[`InputWithFix Component Unit Testing Test InputWithFix Cases 2`] = `
<div
  className="relative my-auto"
>
  <div
    className="absolute inset-y-0 left-0 flex items-center px-2"
  >
    <label
      className=" rounded font-medium px-2 pt-2.5 text-sm dark:text-white cursor-pointer z-30"
    >
      $
    </label>
  </div>
  <input
    className="form-input pl-14 undefined"
    name="test2"
    pattern="[\\\\d\\\\.]{1,}"
    placeholder="0.00"
    required={true}
    type="amount"
  />
</div>
`;

exports[`InputWithFix Component Unit Testing Test InputWithFix Cases 3`] = `
<div
  className="relative my-auto"
>
  <input
    className="form-input undefined pr-14"
    name="percent"
    pattern="^(\\\\d{0,2}(\\\\.\\\\d+)?|100(\\\\.0+)?)$"
    placeholder="0.00"
    required={true}
    type="text"
  />
  <div
    className="absolute inset-y-0 right-0 flex items-center px-2 "
  >
    <label
      className=" font-medium rounded px-2 pt-2.5 text-sm text-gray-600 z-50"
    >
      %
    </label>
  </div>
</div>
`;

exports[`InputWithFix Component Unit Testing Test InputWithFix Cases 4`] = `
<div
  className="relative my-auto"
>
  <div
    className="absolute inset-y-0 left-0 flex items-center px-2"
  >
    <label
      className=" rounded font-medium px-2 pt-2.5 text-sm dark:text-white cursor-pointer z-30"
    >
      $
    </label>
  </div>
  <input
    className="form-input pl-14 pr-14"
    name="amount"
    pattern="[\\\\d\\\\.]{1,}"
    placeholder="00"
    required={true}
    type="text"
  />
  <div
    className="absolute inset-y-0 right-0 flex items-center px-2 "
  >
    <label
      className=" font-medium rounded px-2 pt-2.5 text-sm text-gray-600 z-50"
    >
      .00
    </label>
  </div>
</div>
`;
